tools/setup.py

```diff
@@ -0,0 +1,56 @@
+import subprocess
+import os
+import platform
+from tools.colors import Colors
+
+class SceSetupTool:
+    os = ""
+    color = Colors()
+
+    def check_installation(self, name, command, link):
+        devnull = open(os.devnull, 'wb')
+        software = subprocess.check_call(command, stdout=devnull, stderr=subprocess.STDOUT)
+        if software == 0:
+            self.color.print_yellow(name + " found", True)
+        else:
+            self.color.print_red(name + " not found", True)
+            print("visit here to install: ")
+            self.color.print_purple(link, True)
+            print("press enter to continue: ")
+            choice = input()
+            while choice != "":
+                choice = input()
+
+    def check_os(self):
+        self.os = platform.system()
+        self.color.print_purple(self.os, True)
+
+    def check_directory(self, name):
```

Comment:
similar to the comment about installation checks, create functions like `setup_core_v4` and `setup_discord_bot` that in turn call `check_directory`